tests/L0/run_transformer/test_fused_softmax.py

```diff
@@ -216,5 +224,69 @@ def test_autocast_fused_upper_triangle_mask_softmax(self):
                 actual.backward(g0)
                 expected.backward(g1)
 
+
+class TestGenericFusedSoftmaxKernel(common_utils.TestCase):
+
+    def _setup_qk(self):
+        qlen = [1, 2, 1234, 2322, 2348]
+        klen = [1, 2, 3, 4, 5, 8, 10, 11, 13, 128, 256, 1200, 1234, 2048, 3123, 4096, 4128, 7234, 8192]
+        return itertools.product(qlen, klen)
+
+    def _setup_batch_attn_scalar(self):
+        batch = 2
+        attn = 16
+        scale_t = 1.0
+        return batch, attn, scale_t
+
+    def test_forward(self):
+        import generic_scaled_masked_softmax_cuda
```

Comment:
out of curiosity, why do you call the extension's forward and backward directly instead of `generic_scaled_masked_softmax` defined in `fused_softmax.py`?

Reply:
Either way works. As a unit test, I am focusing testing the low lever kennel implementation.